src/app/content/highlights/reducer.ts

```diff
@@ -194,7 +194,16 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
       };
     }
     case getType(actions.focusHighlight): {
-      return {...state, currentPage: { ...state.currentPage, hasUnsavedHighlight: false, focused: action.payload }};
+      const hasNewFocus = state.currentPage.focused !== action.payload;
+      return {
+        ...state,
+        currentPage: {
+          ...state.currentPage,
+          focused: action.payload,
+          hasUnsavedHighlight: hasNewFocus ? false : state.currentPage.hasUnsavedHighlight,
```

Comment:
@TomWoodward good catch on my assumption here. I also don't think it makes sense to dispatch clearFocusedHighlight before this action - it removes the `focused` property entirely from `state.currentPage` which I don't think we want when shifting focus from one highlight to another.